### DISCUSSION TITLE
Personal "Global" Emotes

### DISCUSSION THREAD

**Discussion by Pronomer122** [SevenTV/Extension#536] [Ideas]
Give each user 25-50 slots (more if subscribed, or more in general if anything) where they add any 7TV emote to (static and GIF alike) for use in any and all twitch channels. These emotes can be seen by both the viewer who used it and everyone else who has 7TV installed (of course, only the viewer can use it because it's *their* personal "global" emote). An optional part to this idea would be to add the ability to rename added emotes to anything the user wants [as long as it's TOS-friendly, of course] and when the emote is hovered over with the mouse, both the edited name and original name can be seen.

Theoretically, this should overall increase traffic to a lot of underrated emotes (spawn new memes in communities, give emote creators more of a easy, simple way to show off their emotes) and quell any unfortunate incidents of accidentally typing out the emote you want and only to discover you're on the wrong stream. And with traffic increase to emotes, it should be possible to click on the emotes in question and be able to either add them from the chat directly into your own list or the streamer themselves to add it to the chat [and edit it later on the 7TV site].

Now, there's definitely a lot of room for bad actors here but a good start would be to completely disable the ability to have "NSFW" emotes for this feature, since no one would want to see a lewd gif emote on a Roblox stream. There's probably other things to consider but I can't think of them from the top of my head, and anyone free to give more examples can provide them.

**Comment by Melonify** [comment]
You'll be happy to hear that this feature already exists! Any 7TV subscriber can choose up to 5 (more may be allowed in the future) Personal Emotes, subject to slightly higher moderation standards, for use in any chat globally on Twitch.

Just subscribe on [7tv.app](https://7tv.app/subscribe), and add emotes to your "Personal Emotes" set, emotes which have not yet been approved for personal use may take some time to be verified, so you can filter on the search page to see only emotes which have already been checked.

![image](https://user-images.githubusercontent.com/13710061/233863785-95325976-50dc-4b48-a744-ce7bc31bca98.png)
![image](https://user-images.githubusercontent.com/13710061/233863814-36b47bc6-4047-4b4c-a679-5c5d01ead7db.png)
